week1/solutions/q3.txt

```diff
@@ -0,0 +1,7 @@
+A Bash script is a text file which contain commands that are executed by the Bash shell.The Bash shell is a command-line interpreter that provides interface to interact with unix-based operating system like linux 
+First line of bash script start from '#!/bin/bash'.
+Bash script has commands like terminal.
+Bash script can use loops and conditions like if else.
+# is used to comment which means that line will not be executed
+.sh is used to create a file for example bash.sh
+
```

Comment:
what are the produnction level use cases of bash script, please mention that as well?

Reply:
1. Automated Backups 
2. Deployment Automation 
3. System Monitoring and Alerts
4. Log management 
5. Scheduled Jobs
6. System configuration and management 
7. Data processing and ETL
8. Security options 
9. Continuous Integration 
10. Infrastructure as Code 
11. Incident Response 
12. Data synchronisation 
13. Custom Application Management 
14. User and Group Management 
15. Maintenance and Cleanup task